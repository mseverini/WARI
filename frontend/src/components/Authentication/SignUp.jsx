import React from 'react'
import {Redirect} from 'react-router-dom'
import {graphql} from 'react-apollo'
import gql from 'graphql-tag'
import SignUpForm from './SignUpForm'

const signUpMutation = gql`
  mutation SignUpUser($credentials: AuthProviderCredentials) {
    createUser(credentials: $credentials)
    {token user {name email}}
  }`


class SignUp extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      email: '',
      password: '',
      mutate: props.mutate,
      loggedIn: false,
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange(e) {
    const target = e.target
    const value = target.type === 'checkbox' ? target.checked : target.value
    const name = target.name
    let updates = {
      [name]: value,
    }

    this.setState(updates)
  }


  handleSubmit(e) {
    e.preventDefault()
    const credentials = {
      name: this.state.name,
      email: this.state.email,
      password: this.state.password,
    }
    this.state.mutate({
      variables: {
        credentials
      },
    }).then(response => {
      let token = response.data.createUser.token
      global.sessionStorage.setItem('token', token)
      this.setState({
        loggedIn: true,
      })
    })
  }

  render() {
    if (this.state.loggedIn) return <Redirect to="/"/>
    return (<SignUpForm handleSubmit={this.handleSubmit} handleChange={this.handleChange}/>)
  }
}

const SignUpWithMutation = graphql(signUpMutation)(SignUp)
export default SignUpWithMutation
