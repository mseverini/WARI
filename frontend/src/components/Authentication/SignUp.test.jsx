import React from 'react'
import {shallow} from 'enzyme'
import SignUp from './SignUp'

test("Renders the SignUp component", ()=>{
  let actual = shallow(<SignUp/>)
  expect(actual).toMatchSnapshot()
})

