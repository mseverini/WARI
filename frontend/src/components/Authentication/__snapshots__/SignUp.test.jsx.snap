// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the SignUp component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SignUp
    mutate={[Function]}
/>,
  "nodes": Array [
    <SignUp
      mutate={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Apollo(SignUp) />,
    "_instance": GraphQL {
      "context": Object {},
      "dataForChildViaMutation": [Function],
      "previousData": Object {},
      "props": Object {},
      "refs": Object {},
      "renderedElement": <SignUp
        mutate={[Function]}
/>,
      "setWrappedInstance": [Function],
      "shouldRerender": false,
      "state": Object {},
      "type": 1,
      "updater": Updater {
        "_renderer": [Circular],
      },
      "version": 0,
    },
    "_newState": null,
    "_rendered": <SignUp
      mutate={[Function]}
/>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Apollo(SignUp) />,
}
`;
